presto-hive/src/main/java/com/facebook/presto/hive/s3/PrestoS3FileSystem.java

```diff
@@ -490,7 +506,8 @@ public boolean mkdirs(Path f, FsPermission permission)
         ListObjectsRequest request = new ListObjectsRequest()
                 .withBucketName(getBucketName(uri))
                 .withPrefix(key)
-                .withDelimiter(PATH_SEPARATOR);
+                .withDelimiter(recursive && filterDirectories ? null : PATH_SEPARATOR)
```

Comment:
could we add testcases for different behaviors of recursive and filterDirectories?

Reply:
Test case added